core/chains/evm/gas/block_history_estimator.go

```diff
@@ -187,7 +187,9 @@ func (b *BlockHistoryEstimator) GetLegacyGas(_ []byte, gasLimit uint32, maxGasPr
 		return nil, 0, errors.New("BlockHistoryEstimator is not started; cannot estimate gas")
 	}
 	if gasPrice == nil {
-		return nil, 0, errors.New("BlockHistoryEstimator has not finished the first gas estimation yet, likely because a failure on start")
+		b.logger.Warnw(fmt.Sprintf("BlockHistoryEstimator couldn't provide an estimation. This is likely because of a failure on start," +
```

Comment:
I remember now why we made it an error and not a warning before.

You have the scenario where chainlink boots but there is some RPC problem that prevents the BlockHistoryEstimator from requesting blocks during startup. In this case we want to block gas bumping with errors until we can get an estimate.

I'm not sure if we want to change that behaviour.

Reply:
I think the right way to do this would be to return error if you have insufficient number of blocks in the history vs. log warning if you have sufficient blocks but no suitable transactions.